docs/references/architecture/adr-115-predictable-block-times.md

```diff
@@ -0,0 +1,133 @@
+# ADR 115: Predictable Block Times
+
+## Changelog
+
+ - April 30 2024: Created by @melekes
+ - May 13 2024: Updated by @melekes
+
+## Status
+
+**Proposed**
+
+## Context
+
+Let's say you're an ABCI application developer and you want to have
+constant block times: 1 block each 6s. How would you do that?
+
+You found out that proposing a block and voting on it takes roughly 2s in your
+network. You instruct validators in your network to change `timeout_commit`
+(_how long a node waits for additional precommits after committing a block,
+before starting on the next height_) from 1s to 4s in the node's config file.
+
+Do you have predictable block times now?
+
+No - you don't. The expected block time will be around 6s, but even in
+favorable runs it will drift far apart from 6s due to:
+
+1. CometBFT going into multiple rounds of consensus (it happens rarely but
+   still).
+2. Network latency.
+3. Clock drifts.
+4. The delay for processing a block.
+
+A validator could also change `timeout_commit` to 0s in its node's config file.
+That means whenever THIS validator proposes a block, the block time will be 2s
+(not 6s!).
+
+Note that the value of `timeout_commit` is static and can't be changed
+without restarting the node.
+
+Because 1-3 are out of your (and our) control, **we can't have constant block
+times**. But we can design a mechanism so that the medium-to-long term average
+block time converges to a desired value.
+
+To achieve that, we need to define a form of variable block time. Namely, if a
+block takes longer than expected, we should be able to render the next block(s)
+faster in order to converge to the desired (average) block time.
+
+### Use Cases
+
+In case of Osmosis, committing a big block is expected to take longer than
+usual. If we know the approximate size of the next block, we could increase
+`timeout_commit` dynamically (if such feature existed) to give the state
+machine some extra time to finish execution.
```

Comment:
Agree. I copied this from https://github.com/tendermint/tendermint/issues/5911#issuecomment-760968413. @ValarDragon @sunnya97 can you please clarify what you're hoping to archive in https://github.com/tendermint/tendermint/issues/5911?

Reply:
Because `timeout_commit` encompasses the block execution.

So, faster nodes will start next height (proposing blocks or expecting for proposals) before slower nodes.